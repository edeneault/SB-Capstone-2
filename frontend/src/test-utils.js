import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";

// create a customRender that wraps the UI in a memory Router

const MemoryRouterWithInitialRoutes = ({ children, initialRoutes }) => {
  return <MemoryRouter initialEntries={initialRoutes}>{children}</MemoryRouter>;
};

const customRender = (ui, options) => {
  const initialRoutes =
    options && options.initialRoutes ? options.initialRoutes : ["/"];

  return render(ui, {
    wrapper: (args) =>
      MemoryRouterWithInitialRoutes({
        ...args,
        initialRoutes,
      }),
    ...options,
  });
};

// re-export everything
export * from "@testing-library/react";

// override render method
export { customRender as render };
