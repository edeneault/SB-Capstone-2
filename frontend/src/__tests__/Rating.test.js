import React from "react";
import { Provider } from "react-redux";
import configureMockStore from "redux-mock-store";
import thunk from "redux-thunk";
import { initialState, render } from "../test-utils";

import Rating from "../components/Rating";

const mockStore = configureMockStore([thunk]);

describe("test", () => {
  let value, text;
  beforeEach(() => {
    (value = 5), (text = `1 reviews`);
  });

  test("renders Rating component without crashing", () => {
    const store = mockStore(initialState);

    const { getByText } = render(
      <Provider store={store}>
        <Rating value={value} text={text} />
      </Provider>,
    );

    expect(getByText("1 reviews")).toBeInTheDocument();
  });

  test("matches snapshot", function () {
    const store = mockStore(initialState);
    const { asFragment } = render(
      <Provider store={store}>
        <Rating value={value} text={text} />
      </Provider>,
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
