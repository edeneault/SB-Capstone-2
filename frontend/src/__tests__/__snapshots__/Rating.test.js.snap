// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="rating"
  >
    <span>
      <i
        class="fas fa-star"
        style="color: rgb(140, 61, 61);"
      />
    </span>
    <span>
      <i
        class="fas fa-star"
        style="color: rgb(140, 61, 61);"
      />
    </span>
    <span>
      <i
        class="fas fa-star"
        style="color: rgb(140, 61, 61);"
      />
    </span>
    <span>
      <i
        class="fas fa-star"
        style="color: rgb(140, 61, 61);"
      />
    </span>
    <span>
      <i
        class="fas fa-star"
        style="color: rgb(140, 61, 61);"
      />
    </span>
    <span>
       1 reviews
    </span>
  </div>
</DocumentFragment>
`;
